apps/server/src/modules/oauth/uc/oauth.uc.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable } from '@nestjs/common';
+import { ILogger, Logger } from '@src/core/logger';
+import axios, { AxiosResponse } from 'axios';
+import jwtDecode from 'jwt-decode';
+import { SystemRepo } from '@shared/repo/system';
+import { UserRepo } from '@shared/repo';
+import { System, User } from '@shared/domain';
+import { FeathersJwtProvider } from '@src/modules/authorization/feathers-jwt.provider';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { Payload } from '../controller/dto/payload';
+import { OauthTokenResponse } from '../controller/dto/oauthTokenResponse';
+import { AuthorizationQuery } from '../controller/dto/authorization.query';
+import { OAuthResponse } from '../controller/dto/oauth.response';
+
+@Injectable()
+export class OauthUc {
+	private logger: ILogger;
+
+	constructor(
+		private readonly systemRepo: SystemRepo,
+		private readonly userRepo: UserRepo,
+		private readonly jwtService: FeathersJwtProvider
+	) {
+		this.logger = new Logger(OauthUc.name);
+	}
+
+	// 0- start Oauth Process
+	async startOauth(query: AuthorizationQuery, systemId: string): Promise<OAuthResponse> {
+		try {
```

Comment:
try catch not required

Reply:
we throw errors, we have to do it that way to send a redirect to the frontend and show the user the right error message. If you have another suggestion or a better way to do it, we are open to suggestions (this will not stay like that for next release, because there will be more than one error message that is sent to the frontend depending on the error).

Reply:
that might happen in the controller... when an error occurs, the controller might redirect to the login form, adding an error code via query which tells the client to show a generic/specific information that the oauth login failed. the use case does not know about rest...

Reply:
Ok done.